test/production/middleware-typescript/app/pages/rewrites/_middleware.ts

```diff
@@ -4,15 +4,12 @@ export const middleware: NextMiddleware = async function (request) {
   const url = request.nextUrl
 
   if (url.pathname === '/') {
-    let bucket = request.cookies.get('bucket')
+    let bucket = request.cookies.get('bucket')?.bucket
```

Comment:
I think `cookies.get("key")` should return the _value_ of the cookies. So `get` returns a `string | null`. Would be happy to know what @chibicode thinks about it.

`getConfig`, `getParsedData`, etc could be handy. Maybe the following could help avoid a leaky abstraction?

```ts
const [value, config] = cookies.getWithConfig("key")
```

Reply:
> I think `cookies.get("key")` should return the value of the cookies. So `get` returns a `string | null`.

+1 on this (I think you meant `string | undefined` instead of `string | null`)

Reply:
@Schniz , In your api, when the raw cookie is set as `foo=barz; Path=/` your api `cookies.get("foo") === 'barz'`, correct?